src/pages/MainPage.js

```diff
@@ -1,10 +1,103 @@
+import styled from '@emotion/styled'
 import { Header, Banner } from '@components'
+import { useState, useEffect } from 'react'
+import { getChannelList, getPostListInChannel, getChannelInfo, getSearchedBookList } from '@apis'
+import { BookListSlider, Input, Button, Select } from '@components'
+
+const ALL_CATEGORY = 'all'
+const DEFAULT_CATEGORY = '소설'
+
+const SearchBar = styled.div`
+  display: flex;
+  justify-content: center;
+`
 
 const MainPage = () => {
+  const [postList, setPostList] = useState([])
+  const [categoryName, setCategoryName] = useState(ALL_CATEGORY)
+
+  const [searchKeyword, setSearchKeyword] = useState('')
+
+  const fetchAllPost = async () => {
+    const channelList = await getChannelList()
+    const totalPostList = await Promise.all(
+      channelList.map(async (channel) => await getPostListInChannel(channel._id))
+    ).then((res) => res.flat())
+
+    totalPostList.sort((post1, post2) => {
+      return Date.parse(post2.createdAt) - Date.parse(post1.createdAt)
+    })
+
+    setPostList(totalPostList)
+  }
+
+  const fetchChannelPost = async (channelName) => {
+    const channelInfo = await getChannelInfo(channelName)
+    const channelPostList = await getPostListInChannel(channelInfo._id)
+    setPostList(channelPostList)
+  }
+
+  // TODO: 상세 포스트 모달 띄우는 로직
+  const handleOnClick = (post) => {
+    console.log(post)
+  }
+
+  useEffect(() => {
+    switch (categoryName) {
+      case ALL_CATEGORY:
+        fetchAllPost()
+        break
+      case DEFAULT_CATEGORY:
+        fetchChannelPost(DEFAULT_CATEGORY)
+        break
+      default:
+        setPostList([])
+    }
+  }, [categoryName])
+
+  useEffect(() => {
+    fetchAllPost()
+  }, [])
+
+  const onChangeSearchKeyword = (e) => {
+    setSearchKeyword(e.target.value)
+  }
+
+  const onSearch = async (e) => {
+    if (!searchKeyword) return
+    const searchResult = await getSearchedBookList(searchKeyword)
+    const searchBookResult = searchResult.filter((result) => !result.role)
```

Comment:
```suggestion
    const searchedResult = await getSearchedBookList(searchKeyword)
    const searchedBookResult = searchResult.filter((result) => !result.role)
```

위처럼 변수명은 명사로 짓는 것이 클린코드로서 추천하는 방법이기 때문에 수정해보는 것이 어떨까요?!

Reply:
꼼꼼한 코드 리뷰 감사합니다!
수경님의 클린한 코드를 걸러내는 필터력을 배우고 싶네요❤